src/BUILD.gn

```diff
@@ -49,16 +49,21 @@ if (chip_build_tests) {
       "${chip_root}/src/inet/tests",
       "${chip_root}/src/lib/asn1/tests",
       "${chip_root}/src/lib/core/tests",
-      "${chip_root}/src/lib/dnssd/minimal_mdns/core/tests",
-      "${chip_root}/src/lib/dnssd/minimal_mdns/responders/tests",
-      "${chip_root}/src/lib/dnssd/minimal_mdns/tests",
-      "${chip_root}/src/lib/dnssd/tests",
       "${chip_root}/src/messaging/tests",
       "${chip_root}/src/protocols/bdx/tests",
       "${chip_root}/src/protocols/user_directed_commissioning/tests",
       "${chip_root}/src/transport/retransmit/tests",
     ]
 
+    if (current_os != "mbed") {
+      deps += [
+        "${chip_root}/src/lib/dnssd/minimal_mdns/core/tests",
```

Comment:
why do tese fail on mbed? I believe at least some ttests here are not using inet at all.

Reply:
If this is due to insufficient memory, please add a note. However I am still surprised if this is the case and maybe we can fix how our unit tests behave (I believe we assumed until now unit tests run on large systems so we allocated more memory, however the constructs in minmdns are simple enough to be testable on embedded)

Reply:
I expect https://github.com/project-chip/connectedhomeip/blob/master/src/lib/dnssd/minimal_mdns/core/tests/TestQName.cpp for example to fit in ram.

Reply:
The issue is an insufficient amount of flash memory on the device which we used. We noticed that when KeyValueStorage is tested. We decided to skip the dnssd tests at all to save the memory. They seem less significant from the perspective of the entire platform.

Reply:
@andy31415 Flash is the issue, not RAM. Unlike tests on the hosts, all tests are combined in a single executable. The binary is large and cannot accommodate flash pages for the non persistent storage.